src/inmanta/agent/agent_new.py

```diff
@@ -157,6 +158,21 @@ async def set_state(self, agent: Optional[str], enabled: bool) -> Apireturn:
                 await self.scheduler.refresh_agent_state_from_db(name=agent)
                 return 200, f"Agent `{agent}` has been notified!"
 
+    async def load_environment_settings(self) -> None:
```

Comment:
I have no strict objection, but might it not be better to simply read the environment setting in `ResourceScheduler.load_timer_settings()`? Optionally in scheduler stage 3.